crates/sui-indexer-alt-restorer/src/snapshot.rs

```diff
@@ -0,0 +1,188 @@
+        );
+        Ok(())
+    }
+
+    async fn restore_object_infos(
+        &self,
+        input_files: Vec<(u32, (u32, FileMetadata))>,
+        epoch_dir: Path,
+        remote_object_store: Arc<dyn ObjectStoreGetExt>,
+        sha3_digests: Arc<Mutex<DigestByBucketAndPartition>>,
+        num_part_files: usize,
+    ) -> anyhow::Result<()> {
+        let move_object_progress_bar = Arc::new(self.snapshot_reader.get_multi_progress().add(
+            ProgressBar::new(num_part_files as u64).with_style(
+                ProgressStyle::with_template(
+                    "[{elapsed_precise}] {wide_bar} {pos} out of {len} move object files restored ({msg})",
+                )
+                .unwrap(),
+            ),
+        ));
+
+        futures::stream::iter(input_files)
+            .try_for_each_spawned(
+                self.restore_args.concurrency,
+                |(bucket, (part_num, file_metadata))| {
+                    let epoch_dir = epoch_dir.clone();
+                    let remote_object_store = remote_object_store.clone();
+                    let sha3_digests = sha3_digests.clone();
+                    let bar = move_object_progress_bar.clone();
+                    let db = self.db.clone();
+                    let next_cp = self.next_checkpoint_after_epoch;
+
+                    async move {
+                        let mut conn = db.connect().await?;
+                        let object_file_path = file_metadata.file_path(&epoch_dir);
+                        let (bytes, _) = download_bytes(
+                            remote_object_store,
+                            &file_metadata,
+                            epoch_dir,
+                            sha3_digests,
+                            &&bucket,
+                            &part_num,
+                            Some(512), // max_timeout_secs
+                        )
+                        .await;
+                        info!(
+                            path = ?object_file_path,
+                            "Finished downloading move object file"
+                        );
+                        let mut object_infos = vec![];
+                        LiveObjectIter::new(&file_metadata, bytes.clone()).map(|obj_iter| {
+                            for object in obj_iter {
+                                match object {
+                                    LiveObject::Normal(obj) => {
+                                        let object_info =
+                                            StoredObjInfo::from_object(&obj, next_cp as i64);
+                                        object_infos.push(object_info);
+                                    }
+                                    LiveObject::Wrapped(_) => {}
+                                }
+                            }
+                        })?;
+                        let object_infos =
+                            object_infos.into_iter().collect::<Result<Vec<_>, _>>()?;
```

Comment:
I think this can be simplified a bit more, but using `?` earlier (rather than `map`-ping over the `Result`) and using `filter_map` to deal with the fact that we want to ignore wrapped objects.

```suggestion
                        let object_infos = LiveObjectIter::new(&file_metadata, bytes.clone())?
                           .filter_map(|object| match object {
                               LiveObject::Normal(obj) => {
                                   Some(StoredObjInfo::from_object(&obj, next_cp as i64))
                               },
                               
                               LiveObject::Wrapped(_) => None
                           })
                           .collect::<Result<Vec<_>, _>>()?;
```